crates/language/src/task_context.rs

```diff
@@ -90,8 +90,13 @@ impl ContextProvider for BasicContextProvider {
             task_variables.insert(VariableName::SelectedText, selected_text);
         }
         if let Some(path) = current_file {
-            task_variables.insert(VariableName::File, path);
+            task_variables.insert(VariableName::File, path.clone());
+
+            if let Some(file) = Path::new(&path).file_stem().and_then(|f| f.to_str()) {
+                task_variables.insert(VariableName::Filename, String::from(file));
```

Comment:
I think this is a bit confusing, as it's just the stem portion of the file, without an extension. I'd either change the name of the variable or (preferably) have it include the full name of the file with extension.

Reply:
I will change the name of the `VariableName::Filename` to `VariableName::Stem` because the rust test CLI command seems not to work if you include the extension.

Reply:
That works; an alternative (that's admittedly a bit ugly) is that we could have it be the filename and process the filename to get rid of the extension before invoking cargo (as a part of a task). I don't think that's too bad honestly.

Reply:
Yeah, sounds good. But the only thing I don't like is that if more task gonna need this, they are going to do the same job on each task. Instead, we are doing it once when building the context. 

But if you think like, we give you the filename and if you only want/need the `stem` than just parse it yourself. So we keep it more general.

Reply:
Yep, that sounds good to me; let's have both